src/cyw43_ll.c

```diff
@@ -2127,9 +2127,11 @@ int cyw43_ll_wifi_join(cyw43_ll_t *self_in, size_t ssid_len, const uint8_t *ssid
         #define WL_CHANSPEC_CTL_SB_NONE     WL_CHANSPEC_CTL_SB_LLL
         #define WL_CHANSPEC_BAND_2G        0x0000
         memcpy(buf + 4 + 32 + 20, bssid, 6);
-        cyw43_put_le32(buf + 4 + 32 + 20 + 8, 1); // chanspec_num
-        uint16_t chspec = channel | WL_CHANSPEC_BW_20 | WL_CHANSPEC_CTL_SB_NONE | WL_CHANSPEC_BAND_2G;
-        cyw43_put_le16(buf + 4 + 32 + 20 + 12, chspec); // chanspec_list
+        if (channel != 0 && channel != -1) {
+            cyw43_put_le32(buf + 4 + 32 + 20 + 8, 1); // chanspec_num
+            uint16_t chspec = channel | WL_CHANSPEC_BW_20 | WL_CHANSPEC_CTL_SB_NONE | WL_CHANSPEC_BAND_2G;
+            cyw43_put_le16(buf + 4 + 32 + 20 + 12, chspec); // chanspec_list
```

Comment:
In the `cyw43_write_iovar_n()` call below, should probably have length as `4 + 32 + 20 + 12` when `chanspec_num=0`.

Reply:
Would you mind if I keep it at 70 bytes? It matches what the infineon code does - they always send a struct of 70 bytes and just zero the buffer when not setting a channel.

Reply:
Yes that's fine, let's keep it as-is.